invenio_sequencegenerator/__init__.py

```diff
@@ -22,7 +22,172 @@
 # waive the privileges and immunities granted to it by virtue of its status
 # as an Intergovernmental Organization or submit itself to any jurisdiction.
 
-"""Invenio module for generating sequences."""
+"""Generic minter for report numbers with auto-increment per custom prefix.
+
+Invenio-SequenceGenerator is a secondary component of Invenio, responsible for
+automatically generating report numbers in a safe manner (i.e. making sure
+that every time a client requests a new report number of a specific form, the
+server will provide him with a new *unique* report number).
+
+Initialization
+==============
+First create a Flask application:
+
+>>> from flask import Flask
+>>> app = Flask('myapp')
+
+In order for the following examples to work, you need to work within an
+Flask application context so let's push one:
+
+>>> ctx = app.app_context()
+>>> ctx.push()
+
+Also, for the examples to work we need to create the database and tables (note,
+in this example we use an in-memory SQLite database):
+
+>>> from invenio_db import db
+>>> db.init_app(app)
+>>> db.create_all()
+
+Usage
+=====
+
+Sequence definition
+-------------------
+
+The first thing to do is define your sequence (i.e. the specific form of your
```

Comment:
`The first thing to do is to define` or `The first thing to do is defining` ... ?